core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -173,6 +181,24 @@ object UnitAutomation {
         if (unitDistanceToTiles.isEmpty()) return true // can't move, so...
         val currentUnitTile = unit.getTile()
 
+        // I'm sure this doesn't kill performance /s
+        val tilesWithRangedEnemyUnits = unit.currentTile.getTilesInDistance(3)
+                .map { it.getUnits().filter { unit.civInfo.isAtWarWith(it.civInfo) }.toList() }
+                .toList().flatten()
+
+        var tilesInRangeOfAttack = tilesWithRangedEnemyUnits.map {
+            it.getTile().getTilesInDistance(it.getRange()).toList()
+        }.flatten()
+
+        // Same with this
+        val tilesWithinBomardmentRange = unit.currentTile.getTilesInDistance(3)
+                .filter { it.isCityCenter() }
+                .filter { it.getCity()?.civInfo!!.isAtWarWith(unit.civInfo) }
+                .map { it.getTilesInDistance(it.getCity()!!.range).toList() }
+                .toList().flatten()
+
+        tilesInRangeOfAttack = ((tilesInRangeOfAttack union tilesWithinBomardmentRange).toList())
```

Comment:
Once these are both sequences, you can just `tilesInRangeOfAttack + tilesWithinBomardmentRange`